### PR TITLE
Make cancelled appeal obvious in case search

### LINKED ISSUE #13654: Make cancelled appeals more obvious in appeal search
## Description
UI problem

At the Board's request, two appeals were modified so that the users will not be able to see or search for them based on the docket number. 
(The original request from Martina was to delete them so that users are not confused about cancelled appeals and their assignment to users -- [screenshots in slack](https://dsva.slack.com/archives/CHX8FMP28/p1583761683209100?thread_ts=1583267009.143600&cid=CHX8FMP28))
* Martina: "This makes it appear its still active and assigned to a user to work" 

Cancelled appeals should be made more obvious in the UI.

Needs Product, Designer, and end-user input.

Per end-user: Show the appeal in search, but show the status as "Cancelled"

## Acceptance criteria
- [ ] Distinguish cancelled appeals in the Case Search Results page
  - [ ] Do not display “assigned to” and indicate that appeal is cancelled/remove/what have you
  - [ ] Display appeals with no issues associated as "cancelled" in the status column of the search page
- [ ] Appeal is still clickable to be able to view the Case Details page

## Background/context/resources
* https://dsva.slack.com/archives/CJL810329/p1583769022223100
* https://dsva.slack.com/archives/CHX8FMP28/p1583764614216100

## Design
Search Cases_Cancelled Appeals Mockup

### PR DESCRIPTION
**Pull request by hschallhorn**
Resolves #13654

### Description
Pass ama appeal status to the front end to be shown in case search. Do not show appeal assignee if it is cancelled. Color status red for cancelled appeals.

### Acceptance Criteria
- [x] No assignee is shown for cancelled appeals in case search
- [x] Cancelled appeals show a red status of cancelled appeals in case search

### Testing Plan
1. Navigate to http://localhost:3000/search?veteran_ids=377
1. Cancel the appeal
```ruby
Appeal.find_by(uuid: "a7654f2e-3aa1-4efa-b671-ce642d43e455").tasks.each(&:cancelled!)
```
1. Ensure these is no assignee listed
1. Ensure the case status is "cancelled" and red

### User Facing Changes

Before|After
---|---
Screen Shot 2020-04-09 at 12 46 58 PM|Screen Shot 2020-04-09 at 12 45 53 PM

### CODE REVIEW

File: app/models/serializers/work_queue/appeal_serializer.rb [hunk 1]
```diff
@@ -21,6 +21,8 @@ class WorkQueue::AppealSerializer
     end
   end
 
+  attribute :status
+
```
Comment (hschallhorn) [inline app/models/serializers/work_queue/appeal_serializer.rb:25]: Ensures we have the status of an ama appeal on the front end

File: client/app/queue/CaseListTable.jsx [hunk 2]
```diff
@@ -3,37 +3,17 @@ import React from 'react';
...
-  }
-
-  const regex = new RegExp(`\\b(?:BVA|VACO|VHAISA)?${assignedToLocation}\\b`);
-
-  if (userId.match(regex) !== null) {
-    return <span {...currentAssigneeStyling}>{COPY.CASE_LIST_TABLE_ASSIGNEE_IS_CURRENT_USER_LABEL}</span>;
-  }
-
-  return assignedToLocation;
-};
```
Comment (hschallhorn) [inline client/app/queue/CaseListTable.jsx:35]: Moved to utils

File: client/app/queue/CaseListTable.jsx [hunk 3]
```diff
@@ -44,13 +24,12 @@ class CaseListTable extends React.PureComponent {
...
-              appeal={appeal}
-              userRole={this.props.userRole}
-              getLinkText={() => appeal.docketNumber} />
-          </React.Fragment>;
+          return (
+            <React.Fragment>
+              <DocketTypeBadge name={appeal.docketName} number={appeal.docketNumber} />
+              <CaseDetailsLink appeal={appeal} userRole={this.props.userRole} getLinkText={() => appeal.docketNumber} />
+            </React.Fragment>
+          );
```
Comment (hschallhorn) [inline client/app/queue/CaseListTable.jsx:32]: Autolinting

File: client/app/queue/CaseListTable.jsx [hunk 4]
```diff
@@ -59,21 +38,19 @@ class CaseListTable extends React.PureComponent {
...
       {
         header: COPY.CASE_LIST_TABLE_APPEAL_TYPE_COLUMN_TITLE,
         valueFunction: (appeal) => renderAppealType(appeal)
       },
       {
         header: COPY.CASE_LIST_TABLE_DECISION_DATE_COLUMN_TITLE,
-        valueFunction: (appeal) => appeal.decisionDate ?
-          <DateString date={appeal.decisionDate} /> :
-          ''
+        valueFunction: (appeal) => (appeal.decisionDate ? <DateString date={appeal.decisionDate} /> : '')
```
Comment (hschallhorn) [inline client/app/queue/CaseListTable.jsx:49]: Autolinting

File: client/app/queue/CaseListTable.jsx [hunk 5]
```diff
@@ -94,21 +71,23 @@ class CaseListTable extends React.PureComponent {
...
+    return (
+      <Table
+        className="cf-case-list-table"
+        columns={this.getColumns}
+        rowObjects={this.props.appeals}
+        getKeyForRow={this.getKeyForRow}
+        styling={this.props.styling}
+      />
+    );
+  };
```
Comment (hschallhorn) [inline client/app/queue/CaseListTable.jsx:90]: Autolinting

File: client/app/queue/CaseListTable.jsx [hunk 6]
```diff
@@ -124,8 +103,15 @@ const mapStateToProps = (state) => ({
...
+    {
+      clearCaseListSearch
+    },
+    dispatch
+  );
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(CaseListTable);
```
Comment (hschallhorn) [inline client/app/queue/CaseListTable.jsx:117]: Autolinting

File: client/app/queue/utils.js [hunk 7]
```diff
@@ -1,14 +1,10 @@
...
+import _, { capitalize, escapeRegExp } from 'lodash';
 import moment from 'moment';
 import StringUtil from '../util/StringUtil';
-import {
-  redText,
-  ISSUE_DISPOSITIONS,
-  VACOLS_DISPOSITIONS,
-  LEGACY_APPEAL_TYPES
-} from './constants';
+import { redText, ISSUE_DISPOSITIONS, VACOLS_DISPOSITIONS, LEGACY_APPEAL_TYPES } from './constants';
```
Comment (hschallhorn) [inline client/app/queue/utils.js:6]: Autolinting

File: client/app/queue/utils.js [hunk 8]
```diff
@@ -29,20 +26,21 @@ import { formatDateStrUtc } from '../util/DateUtil';
...
+    }
 
-  if (issue.disposition && issue.disposition in UNDECIDED_VACOLS_DISPOSITIONS_BY_ID) {
-    return true;
-  }
-});
+    if (issue.disposition && issue.disposition in UNDECIDED_VACOLS_DISPOSITIONS_BY_ID) {
+      return true;
+    }
+  });
```
Comment (hschallhorn) [inline client/app/queue/utils.js:38]: Autolinting

File: client/app/queue/utils.js [hunk 9]
```diff
@@ -29,20 +26,21 @@ import { formatDateStrUtc } from '../util/DateUtil';
...
 
 export const mostRecentHeldHearingForAppeal = (appeal) => {
   const hearings = appeal.hearings.
     filter((hearing) => hearing.disposition === HEARING_DISPOSITION_TYPES.held).
-    sort((h1, h2) => h1.date < h2.date ? 1 : -1);
+    sort((h1, h2) => (h1.date < h2.date ? 1 : -1));
```
Comment (hschallhorn) [inline client/app/queue/utils.js:43]: Autolinting

File: client/app/queue/utils.js [hunk 10]
```diff
@@ -64,10 +62,12 @@ export const prepareMostRecentlyHeldHearingForStore = (appealId, hearing) => {
...
-  const decisionPreparedBy = task.attributes.decision_prepared_by.first_name ? {
-    firstName: task.attributes.decision_prepared_by.first_name,
-    lastName: task.attributes.decision_prepared_by.last_name
-  } : null;
+  const decisionPreparedBy = task.attributes.decision_prepared_by.first_name ?
+    {
+      firstName: task.attributes.decision_prepared_by.first_name,
+      lastName: task.attributes.decision_prepared_by.last_name
+    } :
+    null;
```
Comment (hschallhorn) [inline client/app/queue/utils.js:70]: Autolinting

File: client/app/queue/utils.js [hunk 11]
```diff
@@ -144,27 +144,27 @@ const appealAttributesFromRawTask = (task) => ({
...
+    if (!accumulator[task.attributes.external_appeal_id]) {
+      accumulator[task.attributes.external_appeal_id] = appealAttributesFromRawTask(task);
+    }
 
-      return accumulator;
-    }, {});
-  };
+    return accumulator;
+  }, {});
+};
```
Comment (hschallhorn) [inline client/app/queue/utils.js:155]: Autolinting

File: client/app/queue/utils.js [hunk 12]
```diff
@@ -144,27 +144,27 @@ const appealAttributesFromRawTask = (task) => ({
...
 
-export const extractAppealsAndAmaTasks =
-(tasks) => ({
+export const extractAppealsAndAmaTasks = (tasks) => ({
   tasks: {},
   appeals: extractAppealsFromTasks(tasks),
-  amaTasks: prepareTasksForStore(tasks) });
+  amaTasks: prepareTasksForStore(tasks)
+});
```
Comment (hschallhorn) [inline client/app/queue/utils.js:161]: Autolinting

File: client/app/queue/utils.js [hunk 13]
```diff
@@ -144,27 +144,27 @@ const appealAttributesFromRawTask = (task) => ({
...
 
-export const tasksWithAppealsFromRawTasks = (tasks) => tasks.map((task) => ({
-  ...taskAttributesFromRawTask(task),
-  appeal: appealAttributesFromRawTask(task)
-}));
+export const tasksWithAppealsFromRawTasks = (tasks) =>
+  tasks.map((task) => ({
+    ...taskAttributesFromRawTask(task),
+    appeal: appealAttributesFromRawTask(task)
+  }));
```
Comment (hschallhorn) [inline client/app/queue/utils.js:167]: Autolinting

File: client/app/queue/utils.js [hunk 14]
```diff
@@ -224,17 +224,16 @@ export const prepareAllTasksForStore = (tasks) => {
...
 
-    return {
-      tasks: prepareLegacyTasksForStore(tasks),
-      appeals: extractAppealsFromTasks(tasks)
-    };
+  return {
+    tasks: prepareLegacyTasksForStore(tasks),
+    appeals: extractAppealsFromTasks(tasks)
   };
+};
```
Comment (hschallhorn) [inline client/app/queue/utils.js:236]: Autolinting

File: client/app/queue/utils.js [hunk 15]
```diff
@@ -251,8 +250,8 @@ export const prepareAppealIssuesForStore = (appeal) => {
   return issues;
 };
 
-export const prepareAppealHearingsForStore = (appeal) => appeal.attributes.hearings.
-  map((hearing) => ({
+export const prepareAppealHearingsForStore = (appeal) =>
+  appeal.attributes.hearings.map((hearing) => ({
```
Comment (hschallhorn) [inline client/app/queue/utils.js:254]: Autolinting

File: client/app/queue/utils.js [hunk 16]
```diff
@@ -262,8 +261,8 @@ export const prepareAppealHearingsForStore = (appeal) => appeal.attributes.heari
     isVirtual: hearing.is_virtual
   }));
 
-const prepareAppealAvailableHearingLocationsForStore = (appeal) => appeal.attributes.
-  available_hearing_locations.map((ahl) => ({
+const prepareAppealAvailableHearingLocationsForStore = (appeal) =>
+  appeal.attributes.available_hearing_locations.map((ahl) => ({
```
Comment (hschallhorn) [inline client/app/queue/utils.js:265]: Autolinting

File: client/app/queue/utils.js [hunk 17]
```diff
@@ -275,79 +274,77 @@ const prepareAppealAvailableHearingLocationsForStore = (appeal) => appeal.attrib
...
     };
+
+    return accumulator;
+  }, {});
+
+  return {
+    appeals: appealHash,
+    appealDetails: appealDetailsHash
   };
+};
```
Comment (hschallhorn) [inline client/app/queue/utils.js:347]: Autolinting

File: client/app/queue/utils.js [hunk 18]
```diff
@@ -375,25 +372,34 @@ export const prepareClaimReviewForStore = (claimReviews) => {
...
+  return (
+    <React.Fragment>
+      {isAdvancedOnDocket && (
+        <span>
+          <span {...redText}>AOD</span>,{' '}
+        </span>
+      )}
+      {cavc ? <span {...redText}>CAVC</span> : <span>{caseType}</span>}
+    </React.Fragment>
+  );
```
Comment (hschallhorn) [inline client/app/queue/utils.js:387]: Autolinting

File: client/app/queue/utils.js [hunk 19]
```diff
@@ -375,25 +372,34 @@ export const prepareClaimReviewForStore = (claimReviews) => {
...
+  return (
+    <React.Fragment>
+      {aod && (
+        <span>
+          <span {...redText}>AOD</span>,{' '}
+        </span>
+      )}
+      {cavc ? <span {...redText}>CAVC</span> : <span>{type}</span>}
+    </React.Fragment>
+  );
```
Comment (hschallhorn) [inline client/app/queue/utils.js:402]: Autolinting

File: client/app/queue/utils.js [hunk 20]
```diff
@@ -410,11 +416,7 @@ export const getDecisionTypeDisplay = (checkoutFlow) => {
 export const getIssueProgramDescription = (issue) =>
   _.get(ISSUE_INFO[issue.program], 'description', '') || 'Compensation';
 export const getIssueTypeDescription = (issue) => {
-  const {
-    program,
-    type,
-    description
-  } = issue;
+  const { program, type, description } = issue;
```
Comment (hschallhorn) [inline client/app/queue/utils.js:419]: Autolinting

File: client/app/queue/utils.js [hunk 21]
```diff
@@ -434,9 +436,7 @@ export const getIssueDiagnosticCodeLabel = (code) => {
 };
 
 // Build case review payloads for attorney decision draft submissions as well as judge decision evaluations.
-export const buildCaseReviewPayload = (
-  checkoutFlow, decision, draftDecisionSubmission, issues, args = {}
-) => {
+export const buildCaseReviewPayload = (checkoutFlow, decision, draftDecisionSubmission, issues, args = {}) => {
```
Comment (hschallhorn) [inline client/app/queue/utils.js:439]: Autolinting

File: client/app/queue/utils.js [hunk 22]
```diff
@@ -503,10 +503,7 @@ export const buildCaseReviewPayload = (
  */
 export const validateWorkProductTypeAndId = (decision) => {
   const {
-    opts: {
-      document_id: documentId,
-      work_product: workProduct
-    }
+    opts: { document_id: documentId, work_product: workProduct }
```
Comment (hschallhorn) [inline client/app/queue/utils.js:506]: Autolinting

File: client/app/queue/utils.js [hunk 23]
```diff
@@ -535,8 +532,11 @@ export const taskIsOnHold = (task) => {
 
 export const taskHasCompletedHold = (task) => {
   if (task.onHoldDuration && task.placedOnHoldAt) {
-    return moment().startOf('day').
-      diff(moment(task.placedOnHoldAt), 'days') >= task.onHoldDuration;
+    return (
+      moment().
+        startOf('day').
+        diff(moment(task.placedOnHoldAt), 'days') >= task.onHoldDuration
+    );
```
Comment (hschallhorn) [inline client/app/queue/utils.js:539]: Autolinting

File: client/app/queue/utils.js [hunk 24]
```diff
@@ -577,14 +577,16 @@ export const nullToFalse = (key, obj) => {
 
 export const sortTaskList = (taskList) => {
   return taskList.sort((prev, next) => {
-    return new Date(next.closedAt || next.createdAt).getTime() -
-    new Date(prev.closedAt || prev.createdAt).getTime();
+    return new Date(next.closedAt || next.createdAt).getTime() - new Date(prev.closedAt || prev.createdAt).getTime();
```
Comment (hschallhorn) [inline client/app/queue/utils.js:580]: Autolinting

File: client/app/queue/utils.js [hunk 25]
```diff
@@ -577,14 +577,16 @@ export const nullToFalse = (key, obj) => {
...
 };
 
 export const regionalOfficeCity = (objWithLocation, defaultToUnknown) => {
-  return _.get(objWithLocation, 'closestRegionalOffice.location_hash.city',
-    defaultToUnknown ? COPY.UNKNOWN_REGIONAL_OFFICE : defaultToUnknown);
+  return _.get(
+    objWithLocation,
+    'closestRegionalOffice.location_hash.city',
+    defaultToUnknown ? COPY.UNKNOWN_REGIONAL_OFFICE : defaultToUnknown
+  );
```
Comment (hschallhorn) [inline client/app/queue/utils.js:589]: Autolinting

File: client/app/queue/utils.js [hunk 26]
```diff
@@ -594,7 +596,51 @@ export const cityForRegionalOfficeCode = (code) => {
 };
 
 export const hasDASRecord = (task, requireDasRecord) => {
-  return (task.appeal.isLegacyAppeal && requireDasRecord) ? Boolean(task.taskId) : true;
+  return task.appeal.isLegacyAppeal && requireDasRecord ? Boolean(task.taskId) : true;
```
Comment (hschallhorn) [inline client/app/queue/utils.js:599]: Autolinting

File: client/app/queue/utils.js [hunk 27]
```diff
@@ -594,7 +596,51 @@ export const cityForRegionalOfficeCode = (code) => {
...
+ * Method to determine whether to apply styling to the Current Assignee (location)
+ * @param {object} appeal -- The appeal for which to determine the location
+ * @param {number} userId -- The ID of the current user for showing "Assigned to You" cases
+ * @returns {string} -- The value of the current location either as a string or JSX
+ */
+export const labelForLocation = (appeal, userId) => {
+  // If there is no location or the appeal is cancelled, don't show a location
+  if (!appeal.assignedToLocation || appeal.status === 'cancelled') {
+    return '';
+  }
```
Comment (hschallhorn) [inline client/app/queue/utils.js:614]: Change 1! Do not show "assignee" if the case has been cancelled

File: client/app/queue/utils.js [hunk 28]
```diff
@@ -594,7 +596,51 @@ export const cityForRegionalOfficeCode = (code) => {
...
+};
+
+/**
+ * Method to optionally apply styling to the status column
+ * @param {object} appeal -- The appeal for which to determine the status
+ * @returns {string} -- The value of the current location either as a string or JSX
+ */
+export const statusLabel = (appeal) =>
+  appeal.status === 'cancelled' ? (
+    <span {...css({ color: COLORS.RED })}>{capitalize(appeal.status)}</span>
```
Comment (hschallhorn) [inline client/app/queue/utils.js:643]: Change 2! if the appeal is cancelled, make it RED.

File: client/app/queue/utils.js (hunk 18 above)
Comment (lomky) [reply client/app/queue/utils.js:387]: Whats with the added `{' '}` on 382?

File: client/app/queue/utils.js (hunk 19 above)
Comment (lomky) [reply client/app/queue/utils.js:402]: Whats with the added `{' '}` on 397?

Comment (nanotone) [reply client/app/queue/utils.js:402]: Remember `Built with ♡ by theDigital Service at VA`?

File: client/app/queue/utils.js (hunk 18 above)
Comment (hschallhorn) [reply client/app/queue/utils.js:387]: Maybe to make sure there was not whitespace at the end of the line
```diff
-  </span>, </span>
+. </span>,{' '}
+</span>

Comment (hschallhorn) [reply client/app/queue/utils.js:387]: Yup, just saw Yang's comment below

**Pull request merged**